Xamarin.Forms.Core/Page.cs

```diff
@@ -337,6 +339,16 @@ public void SendDisappearing()
 			EventHandler handler = Disappearing;
 			if (handler != null)
 				handler(this, EventArgs.Empty);
+
+			FindApplication(this)?.OnPageDisappearing(this);
+		}
+
+		Application FindApplication(Element element)
+		{
+			if (element == null)
+				return null;
+
+			return (element.Parent is Application app) ? app : FindApplication(element.Parent);
```

Comment:
you can avoid recursion by using a `while` loop

Reply:
How about using `Application.Current`?